streams/src/main/java/org/apache/kafka/streams/processor/internals/StateManagerUtil.java

```diff
@@ -38,13 +41,39 @@
  */
 final class StateManagerUtil {
     static final String CHECKPOINT_FILE_NAME = ".checkpoint";
+    static final long OFFSET_DELTA_THRESHOLD_FOR_CHECKPOINT = 10_000L;
```

Comment:
This is just a placeholder for making less frequent flushing than commits, but it's open for debate whether we have a better mechanism to decide when flushing should be executed.

Reply:
It would be awesome if we could do it based on the size of the pending data and the configured memtable size. Not sure if that's really feasible, just throwing it out there

Reply:
You mean when the accumulated pending data is larger than the memtable, we should checkpoint?

Reply:
Something like that. We know rocksdb will render the memtable immutable once it reaches the configured memtable size, after that it will flush once the number of immutable memtables reaches the configured value. Probably makes sense to align our checkpoint/flushing to the configured rocksdb flushing.

Would be cool if we could piggy-back on the rocksdb options and avoid a new config in Streams altogether, but obviously not everyone uses rocksdb

Reply:
Yeah my major concern is to tie the flushing policy with rocksdb -- although it is the default persistent stores now, we should avoid tying with a specific type of stores.